server/ldap_check.rb

```diff
@@ -86,6 +86,8 @@ def is_user_in_admin_hash user
       @updateCount += 1
       @lastUpdate = Time.now()
       save_group_members configuration['group']
+      # Might get a group with no members or that doesn't exist.
+      @admin_hash = Hash.new unless(@admin_hash)
```

Comment:
"save_group_members" is just concerned with getting the list of members to save.  It delegates all the hash management to other methods.  There are only two of those and they may be called in either order so they both make sure that the hash exists. It could be organized differently but this approach seems clear enough for a 2 page class.